### PR TITLE
Compare release values and desired values in resourceDiff

### LINKED ISSUE #472: When the release fails for any reason do not update the state
The scenario is that a Helm release fails to deploy resources due to any error. Ie. Pod health check failure or misconfiguration of helm chart. The release updates the tf state and therefor wont redeploy on next attempt. 

### Terraform Version and Provider Version
Terraform v0.12.21
Helm v3.1.2

### Provider Version
provider.helm v1.1.1

### Affected Resource(s)
- helm_release

### Expected Behavior
If any failure occurs during deployment of the release it shouldn't mark the deployment as done

### Actual Behavior
No changes are detected between new release and previous release so terraform doesn't trigger a new deployment.

### PR DESCRIPTION
**Pull request by WeiAnAn**
### Description

Compare release values and desired values in resourceDiff.
It allow us resync the state between helm release and terraform state when they are inconsistent.
It is useful when the upgrade is failed setting `atomic = true` option that mentioned in #472.

Here is the execution result.
image

One side effect is that would always shows the `metadata.values` changes when the values is change, and it may make diff result more longer.

**Note: Currently not support when the changed value is setting by `set_sensitive`, because the value will be formatted to `(sensitive data)`. It is the same in diff. 
We can support `set_sensitive` in later PRs. It can be implemented by set these sensitive values to hash, just like `manifest` does.**

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
Compare release values and desired values in resourceDiff
```
### References

Fix: #472 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: helm/resource_release.go [hunk 1]
```diff
@@ -953,10 +953,45 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
...
+	cloakSetValues(desiredValues, d)
+	desiredValuesJson, err := json.Marshal(desiredValues)
+	if err != nil {
+		return err
+	}
+	desiredValuesJsonString := string(desiredValuesJson)
+
+	if desiredValuesJsonString != d.Get("metadata.0.values").(string) &&
+		!(desiredValuesJsonString == "{}" && d.Get("metadata.0.values").(string) == "null") {
+		err = d.SetNew("metadata", []map[string]interface{}{{
```
Comment (WeiAnAn) [inline helm/resource_release.go:980]: Terraform [does not support to `SetNew` with nested key](https://github.com/hashicorp/terraform-plugin-sdk/issues/459) so I set whole metadata here.

**Comment by WeiAnAn** [comment]
Hi @jrhouston, @arybolovlev just wanted to check in on my pull request that's been open for a while. 
If you have a moment to review it, I would really appreciate it. 
Let me know if there's anything else I can do to help move this along. 
Thanks!

**Comment by thehandsomezebra** [comment]
Is there anything that can be done to accelerate this merge? @jrhouston  @arybolovlev  ?
The issue #472 has been persisting since 2020 and this would make this provider significantly more reliable for teams. 
Teams like mine would prefer to stick with Terraform: but with this bug, we are blocked.

**Comment by pecigonzalo** [comment]
Can we re-open this? Seems quite relevant.

**Pull request closed without merging**